autorest/utility_test.go

```diff
@@ -234,6 +234,29 @@ func TestStringWithValidString(t *testing.T) {
 	}
 }
 
+func TestStringWithStringSlice(t *testing.T) {
+	s := []string{"string1", "string2"}
+	if String(s, ",") != "string1,string2" {
+		t.Fatal("autorest: String method failed to convert string slice to string")
+	}
+}
+
+type TestEnumType string
+
+func TestStringWithEnum(t *testing.T) {
+	s := TestEnumType("string1")
+	if String(s) != "string1" {
+		t.Fatal("autorest: String method failed to convert slice of enums to string")
+	}
+}
+
```

Comment:
In addition to the tests that are here, I'd add an `Example` for the `AsStringSlice` to try to make more clear what it does.

I think this is a really natural candidate for an `Example` function instead of a `test` function. Consider this:

``` Go
func ExampleAsStringSlice() {
    type TestEnumType string

    a := []TestEnumType{"value1", "value2"}
    b, err := AsStringSlice(s)
    for _, c := range b {
        fmt.Println(c)
    }
    // Output:
    // value1
    // value2
}
```